api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -802,6 +802,11 @@ message JwtAuthentication {
   // :ref:`requirement_name <envoy_v3_api_field_extensions.filters.http.jwt_authn.v3.PerRouteConfig.requirement_name>`
   // in ``PerRouteConfig`` uses this map to specify a JwtRequirement.
   map<string, JwtRequirement> requirement_map = 5;
+
+  // A request failing the verification process will receive a 401 downstream with the failure response details
```

Comment:
Couple of questions:
1. Why not use https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/custom_response/v3/custom_response.proto.html here? It was designed for this use case. I'm on the fence, as it seems kind of generally useful to be able to do this, but there are definitely alternatives.
2. Can we future proof the tests below to ensure that only standard Envoy headers that don't leak response details are returned?

Reply:
1) I agree that there are a few ways including the custom response filter to solve this, but AFAIK this filter doesn't have awareness of where the response details are generated from? For eg., if the JWT filter is used in a reverse proxy that fronts multiple sites, it could easily strip the wrong data if we match it based on 401. We can get quite specific with the matcher on the response body and limit it to JWT failure reasons, but it starts to seem a bit hacky IMO (If any new failure reason is added, this would need change). 

In a regular filter, I'd probably lean towards not adding more customization like this and rely on an alternative method, but given this filter is security critical, I personally think it'd be a good addition. Atleast in a usecase I had recently worked on, we are using a custom lua script to handle this currently. Let me know what you think

2) I believe we can. The only header in response that envoy adds after stripping is currently "server":"envoy". We can just do a negative test against all headers except that. I'll add this if you think we can move forward with this change.

Reply:
OK, let's do (2) and I think your point on (1) is fair.

For the benefit of @envoyproxy/api-shepherds I actually think this PR and what's being done in https://github.com/envoyproxy/envoy/pull/34719 are not dissimilar in what they're trying to do - have filter config express some control over the local reply body details. 

@tyxia thoughts on how this might be done more generically without adding material complexity?

Reply:
@tyxia Friendly ping. Happy to make any changes that could help here

Reply:
Thank you for the details. 

To ensure no internal data is leaked, I've modified the test to enforce it. 

Basically the headers that are produced are server [envoy], date, status [401]. HTTP Parser seems to produce ContentLength header as 0, but Balsa doesn't produce it when body size is 0. I've added the test to check that the header list size is 3 or 4 and depending on it, check the content of the headers further to align with our expectations. I've also added a condition on the body being empty. Please let me know if any further additions would be useful